tiny_dnn/layers/global_average_pooling_layer.h

```diff
@@ -109,14 +109,17 @@ class global_average_pooling_layer : public layer {
       core::OpKernelConstruction(layer::device(), &params_);
 
     layer::set_backend_type(backend_type);
-    if (backend_type == backend_t::internal || backend_type == backend_t::avx ||
-        backend_type == backend_t::nnpack) {
-      kernel_fwd_.reset(new GlobalAvePoolOp(ctx));
-      kernel_back_.reset(new GlobalAvePoolGradOp(ctx));
-      return;
-    } else {
-      throw nn_error("Not supported engine: " + to_string(backend_type));
+    if (backend_type == core::backend_t::avx) {
+#ifndef CNN_USE_AVX
+      nn_warn(
+        "tiny-dnn has not been compiled with AVX support, "
+        "fallback to internal backend for global avepool layer.\n");
+      layer::set_backend_type(core::backend_t::internal);
+#endif
```

Comment:
I think removing `backend_t::avx` entry if `CNN_USE_AVX` isn't defined is a better solution.
We don't need to rely static_assert or nn_warn exception.
This is C++ header only library, we can keep it simple.